src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -1699,10 +1722,13 @@ void QueryAnalyzer::evaluateScalarSubqueryIfNeeded(QueryTreeNodePtr & node, size
 
     auto io = interpreter->execute();
 
-    Block block;
     PullingAsyncPipelineExecutor executor(io.pipeline);
     io.pipeline.setProgressCallback(context->getProgressCallback());
 
+    Block block;
+    Field scalar_value;
```

Comment:
Not sure why we need to move `scalar_value` and `scalar_type` at the top of the function.

Reply:
I had moved it above for the previous approach, but now no longer needed. Reverted it.